src/m365/aad/commands/user/user-get.ts

```diff
@@ -37,20 +39,47 @@ class AadUserGetCommand extends GraphCommand {
 
   public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
     const properties: string = args.options.properties ?
-      `?$select=${args.options.properties.split(',').map(p => encodeURIComponent(p.trim())).join(',')}` :
+      `&$select=${args.options.properties.split(',').map(p => encodeURIComponent(p.trim())).join(',')}` :
       '';
 
+    let requestUrl: string = `${this.resource}/v1.0/users`;
+
+    if (args.options.id) {
+      requestUrl += `?$filter=id eq '${encodeURIComponent(args.options.id as string)}'${properties}`;
+    }
+    else if (args.options.userName) {
+      requestUrl += `?$filter=userPrincipalName eq '${encodeURIComponent(args.options.userName as string)}'${properties}`;
+    }
+    else if (args.options.email) {
+      requestUrl += `?$filter=mail eq '${encodeURIComponent(args.options.email as string)}'${properties}`;
+    }
+
     const requestOptions: any = {
-      url: `${this.resource}/v1.0/users/${encodeURIComponent(args.options.id ? args.options.id : args.options.userName as string)}${properties}`,
+      url: requestUrl,
       headers: {
         accept: 'application/json;odata.metadata=none'
       },
       responseType: 'json'
     };
 
     request
-      .get(requestOptions)
-      .then((res: any): void => {
+      .get<{ value: User[] }>(requestOptions)
+      .then((res: { value: User[] }): Promise<User> => {
+        if (res.value.length === 1) {
+          return Promise.resolve(res.value[0]);
+        }
+
+        const identifier = args.options.id ? `id ${args.options.id}`
+          : args.options.userName ? `user name ${args.options.userName}`
+            : `email ${args.options.email}`;
+
+        if (res.value.length === 0) {
+          return Promise.reject(`The specified user with ${identifier} does not exist`);
+        }
+
+        return Promise.reject(`Multiple users with ${identifier} found. Please disambiguate (user names): ${res.value.map(a => a.userPrincipalName).join(', ')} or (ids): ${res.value.map(a => a.id).join(', ')}`);
+      })
+      .then((res: any): any => {
```

Comment:
This line should be:

```ts
 .then((res: User): void => {
```

or for short:

```ts
 .then(res => {
```

Let's avoid using `any` when possible